addons/binding/org.openhab.binding.regoheatpump/src/main/java/org/openhab/binding/regoheatpump/handler/Rego6xxHeatPumpHandler.java

```diff
@@ -0,0 +1,405 @@
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.regoheatpump.internal.protocol.RegoConnection;
+import org.openhab.binding.regoheatpump.internal.rego6xx.CommandFactory;
+import org.openhab.binding.regoheatpump.internal.rego6xx.ErrorLine;
+import org.openhab.binding.regoheatpump.internal.rego6xx.RegoRegisterMapper;
+import org.openhab.binding.regoheatpump.internal.rego6xx.ResponseParser;
+import org.openhab.binding.regoheatpump.internal.rego6xx.ResponseParserFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Rego6xxHeatPumpHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Boris Krivonog - Initial contribution
+ */
+abstract class Rego6xxHeatPumpHandler extends BaseThingHandler {
+
+    private static final class ChannelDescriptor {
+        private Date lastUpdate;
+        private byte[] cachedValue;
+
+        public byte[] cachedValueIfNotExpired(int refreshTime) {
+            if (lastUpdate == null || (lastUpdate.getTime() + refreshTime * 900 < new Date().getTime())) {
+                return null;
+            }
+
+            return cachedValue;
+        }
+
+        public void setValue(byte[] value) {
+            lastUpdate = new Date();
+            cachedValue = value;
+        }
+    }
+
+    private final Logger logger = LoggerFactory.getLogger(Rego6xxHeatPumpHandler.class);
+    private final Map<String, ChannelDescriptor> channelDescriptors = new HashMap<>();
+    private int refreshInterval;
+    private RegoConnection connection;
+    private RegoRegisterMapper mapper;
+    private ScheduledFuture<?> scheduledRefreshFuture;
+
+    protected Rego6xxHeatPumpHandler(Thing thing) {
+        super(thing);
+    }
+
+    protected abstract RegoConnection createConnection();
+
+    @Override
+    public void initialize() {
+        mapper = RegoRegisterMapper.rego600();
+        refreshInterval = ((Number) getConfig().get(REFRESH_INTERVAL)).intValue();
+
+        connection = createConnection();
+        logger.info("Rego controller connected via {}", connection.connectionInfo());
+
+        scheduledRefreshFuture = scheduler.scheduleWithFixedDelay(this::refresh, 2, refreshInterval, TimeUnit.SECONDS);
+
+        updateStatus(ThingStatus.UNKNOWN);
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+
+        connection.close();
```

Comment:
Thanks, added the null check within dispose. Regarding the line 107 - since I'm not a Java dev per see, nor a OH expert, can you please confirm the following:
* within the dispose method, scheduledRefreshFuture is cancelled before connection is closed/nulled
* within the handleDataFromHusdataInterface, local connection is taken and Thread.interrupted is checked after that 
So is my assumption right, that Thread.interrupted will be signaled if connection was (or is to happen) closed within the dispose method? And therefore there is actually no explicit need to check the connection for null in the handleDataFromHusdataInterface? Please don't get me wrong, but would just like to understand how stuff works under the hood.